File: app/src/main/java/com/sppe/shrimppaste/ui/activity/HomeActivity.java
package com.sppe.shrimppaste.ui.activity;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import com.sppe.shrimppaste.R;
import com.sppe.shrimppaste.ui.fragment.AndroidFragment;
import com.sppe.shrimppaste.ui.fragment.GirlFragment;
import com.sppe.shrimppaste.ui.fragment.IosFragment;
import com.sppe.shrimppaste.ui.fragment.PersonalFragment;

/**
 * Created by WHF on 2017/11/5.
 */

public class HomeActivity extends AppCompatActivity {

    private Fragment[] fragmentList;
    private int currentPosition;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        initFragmentList();
        initNavigationView();
    }

    private void initFragmentList() {
        fragmentList = new Fragment[4];
        fragmentList[0] = new GirlFragment();
        fragmentList[1] = new AndroidFragment();
        fragmentList[2] = new IosFragment();
        fragmentList[3] = new PersonalFragment();

        currentPosition = 0;
    }


    private void initNavigationView() {
        BottomNavigationView bottomNavigationView = (BottomNavigationView) findViewById(R.id.bnv_home_bottom);
        getSupportFragmentManager().beginTransaction()
                .add(R.id.rl_home_content, fragmentList[currentPosition]).commit();
        bottomNavigationView.setOnNavigationItemSelectedListener(
                new BottomNavigationView.OnNavigationItemSelectedListener() {
                    @Override
                    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                        onBottomTabItemClick(item);
                        getSupportFragmentManager().beginTransaction()
                                .replace(R.id.rl_home_content, fragmentList[currentPosition]).commit();
                        return true;
                    }
                }
        );
    }

    private void onBottomTabItemClick(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.item_girl:
                currentPosition = 0;
                break;
            case R.id.item_android:
                currentPosition = 1;
                break;
            case R.id.item_ios:
                currentPosition = 2;
                break;
            case R.id.item_more:
                currentPosition = 3;
                break;
            default:
                break;
        }
    }

}

File: app/src/main/java/com/sppe/shrimppaste/ui/activity/HomeActivity.kt
package com.sppe.shrimppaste.ui.activity

import android.support.v4.app.Fragment
import android.os.Bundle
import android.view.MenuItem

import com.sppe.shrimppaste.R
import com.sppe.shrimppaste.ui.fragment.AndroidFragment
import com.sppe.shrimppaste.ui.fragment.GirlFragment
import com.sppe.shrimppaste.ui.fragment.IosFragment
import com.sppe.shrimppaste.ui.fragment.PersonalFragment

import com.sppe.shrimppaste.base.BaseActivity
import kotlinx.android.synthetic.main.activity_home.*

/**
 *
 * Created by WangHaoFei on 2017/10/22.
 */
class HomeActivity : BaseActivity() {

    private var fragmentList: Array<Fragment> = arrayOf(
            AndroidFragment(),
            IosFragment(),
            GirlFragment(),
            PersonalFragment()
    )

    private var currentFragment: Fragment = fragmentList[0]

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        initNavigationView()
    }

    private fun initNavigationView() {
        supportFragmentManager.beginTransaction().add(R.id.rl_home_content, fragmentList[0]).commit()
        bnv_home_bottom.setOnNavigationItemSelectedListener { item ->
            onBottomTabItemClick(item)
            supportFragmentManager.beginTransaction()
                    .replace(R.id.rl_home_content, currentFragment).commit()
            true
        }
    }

    private fun onBottomTabItemClick(item: MenuItem) {
        when (item.itemId) {
            R.id.item_android -> currentFragment = fragmentList[0]
            R.id.item_ios -> currentFragment = fragmentList[1]
            R.id.item_girl -> currentFragment = fragmentList[2]
            R.id.item_more -> currentFragment = fragmentList[3]
        }
    }

}
